src/meta/meta_table.c

```diff
@@ -64,44 +64,22 @@ __wt_metadata_open(WT_SESSION_IMPL *session)
 	if (F_ISSET(btree, WT_BTREE_NO_LOGGING))
 		F_CLR(btree, WT_BTREE_NO_LOGGING);
 
-	/* The metadata handle doesn't need to stay locked -- release it. */
-	return (__wt_session_release_btree(session));
+	return (0);
 }
 
 /*
- * __wt_metadata_cursor --
- *	Opens a cursor on the metadata.
+ * __wt_metadata_session_cursor --
+ *	Opens the session's cached metadata cursor.
  */
 int
-__wt_metadata_cursor(
-    WT_SESSION_IMPL *session, const char *config, WT_CURSOR **cursorp)
+__wt_metadata_session_cursor(WT_SESSION_IMPL *session, WT_CURSOR **cursorp)
```

Comment:
This name change seems wrong to me.  Every cursor is already tied to a session, so adding "session" to the name doesn't make the intent clear.  How about calling this `__wt_metadata_cursor` and the other one `__wt_metadata_cursor_open`?